src/test/java/com/google/api/generator/engine/ast/AnonymousClassExprTest.java

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+import static org.junit.Assert.assertThrows;
+
+import com.google.common.base.Function;
+import java.util.Arrays;
+import org.junit.Test;
+
+public class AnonymousClassExprTest {
+  @Test
+  public void validAnonymousClass_basic() {
+    ConcreteReference ref = ConcreteReference.withClazz(Runnable.class);
+    TypeNode type = TypeNode.withReference(ref);
+    AssignmentExpr assignmentExpr = createAssignmentExpr("foobar", "false", TypeNode.BOOLEAN);
+    ExprStatement statement = ExprStatement.withExpr(assignmentExpr);
+    MethodDefinition method =
+        MethodDefinition.builder()
+            .setScope(ScopeNode.PUBLIC)
+            .setReturnType(TypeNode.VOID)
+            .setName("run")
+            .setIsOverride(true)
+            .setBody(Arrays.asList(statement))
+            .build();
+
+    AnonymousClassExpr anonymousClassExpr =
+        AnonymousClassExpr.builder().setType(type).setMethods(Arrays.asList(method)).build();
+    assertTrue(TypeNode.isReferenceType(anonymousClassExpr.type()));
+    assertEquals(anonymousClassExpr.type(), type);
+  }
+
+  @Test
+  public void validAnonymousClass_genericAndVariableExpr() {
+    // new Function<String, String>()
```

Comment:
Can we make some comment edits so we know this is not commented-out code, but refers to what the AST fragments are trying to construct?

Reply:
sure! added `[Constructing]` to indicate that we are constructing AST fragments.